include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_pressure_correction.cpp

```diff
@@ -318,19 +318,35 @@ void
 TimeIntBDFPressureCorrection<dim, Number>::do_timestep_solve()
 {
   // perform the sub-steps of the pressure-correction scheme
-
-  momentum_step();
+  if(this->update_velocity)
+  {
+    momentum_step();
+  }
 
   VectorType pressure_increment;
-  pressure_increment.reinit(pressure_np, false /* init with zero */);
-
-  pressure_step(pressure_increment);
+  if(this->update_pressure)
+  {
+    pressure_increment.reinit(pressure_np, false /* omit_zeroing_entries */);
+    pressure_step(pressure_increment);
+  }
+  else
+  {
+    AssertThrow(this->store_solution and pressure_increment_last_iter.size() > 0,
+                dealii::ExcMessage("Previous pressure step required to "
+                                   "use `pressure_increment_last_iter`."));
+    pressure_increment = pressure_increment_last_iter;
+  }
 
-  projection_step(pressure_increment);
+  if(this->update_velocity)
+  {
+    projection_step(pressure_increment);
+  }
 
-  // evaluate convective term once the final solution at time
-  // t_{n+1} is known
-  evaluate_convective_term();
+  // evaluate convective term once the final solution at time t_{n+1} is known
+  if(this->update_velocity)
+  {
+    evaluate_convective_term();
+  }
```

Comment:
This change could be dangerous in general. I am not completely sure without looking into the code in detail, but we might (or, right now, I actually think that we) need this step so that the next time step is computed correctly. This would imply that we need it in the last FSI iteration even if `this->update_velocity == false`. We do not know at this point whether this will be the last FSI iteration. So if this reasoning is true, we would have to call this function always.

Reply:
The semi-implicit version in this PR focus on the FSI case only. 
(the fluid solvers are also semi-implicit for some combinations, see `TreatmentOfConvectiveTerm`)
within a fluid solver *alone*, however we always call `do_timestep_solve()` with 
`update_pressure == true` and `update_velocity == true`
as they are initialized with `true`.

So whatever happens in `evaluate_convective_term()` can now only lead to changed behavior in the FSI case.
FSI always needs the ALE formulation. Some people might want to do this, but this combination is asserted for now.
Anyways, this function does nothing for the ALE case:
![image](https://github.com/user-attachments/assets/4d8e4a4b-0726-4484-975a-ccf5ceb7c067)

so I think we should be good.
The question wether or not FSI always requires ALE is I think a different one.